### DISCUSSION TITLE
Ability to install cli without core plugins

### DISCUSSION THREAD

**Discussion by Amit-Tripathi** [forcedotcom/cli#2933] [Ideas]
CLI is ideal for local development and works well. For cloud operations like on AWS Lambda, CLI size often creates issues as it installs all core plugins. Ask is that we should be able to install CLI bare minimum and install plugins like source retrieve or package based on the operations we would like to run.
We do have a workaround by using JSFORCE or creating our own tooling api based solution but we have to convert from source to mdapi format for it to work.

**Comment by VivekMChawla** [comment]
@Amit-Tripathi - Could you specify the core plugins you believe should not be installed as part of a minimal CLI distribution?

**Comment by Amit-Tripathi** [comment]
Thank You @VivekMChawla .
Overall size limit for AWS Function Size is 250 MB including dependencies, not knowing the size of these core plugin, my suggestion would be to only have essential plugins like auth, org , settings and ability to pull in plugins as dependencies via script. 

We may not need below plugins for minimal distribution-
templates, source, sObject, Schema, apex, data, user, telemetry, deploy-retrieve, info, limits, marketplace, packaging, trust.

**Reply by cristiand391** [reply]
how are you installing the CLI?

OS installers/tarballs include a node binary but you could skip that by installing `sf` from npm if your env VM already has node/npm

**Reply by VivekMChawla** [reply]
@Amit-Tripathi - While we look into this, I'd like to suggest a custom tooling option that doesn't require using the Tooling API directly.

Everything the CLI does with both Metadata and SFDX source, including conversion from one format to the other, happens via the [SDR (source-deploy-retrieve) Library](https://github.com/forcedotcom/source-deploy-retrieve).  This can be combined with [SFDX Core](https://github.com/forcedotcom/sfdx-core) to handle auth in a relatively straightforward custom solution.

Check out the [SDR Handbook](https://github.com/forcedotcom/source-deploy-retrieve/blob/main/HANDBOOK.md) to learn more about what's possible.
